### ISSUE TITLE
Customize zoom by display?

### ISSUE THREAD

**Issue by hmijail** [Enchufadoo/fixedzoom#15]
I have 2 displays, one with a higher resolution than the other. MacOS has disappointingly dropped the ball and one of the displays renders everything significantly smaller than the other. 
So I would like to use 100% zoom in one display and 120% zoom in the other.
Could fixedzoom do that?

**Comment by Enchufadoo** [comment]
Someone already had asked in the addon reviews about that, different zoom levels for different dpis.
It's doable but it's complicated, mostly because also would have to create separated rules for different dpis. the dialogs, basically everything has to change at least a bit.

**Comment by Enchufadoo** [comment]
@hmijail I asked in the mozilla forums if there's a way to obtain the screen resolution for the current tab
https://discourse.mozilla.org/t/getting-the-screen-resolution-from-a-background-script/33108

But it's not possible, is not something that the browser shares apparently, it's funny because if you go to a page like this one http://dpi.lv you can get the resolution, but the extensions API doesn't provide a way of knowing this. I could for example insert a script in every tab the user visits, but I think that's not fast enough for the zoom change to work as it does now.

I'll leave the issue open maybe someone else knows how to.

**Comment by saildata** [comment]
I came here to request a related feature -- profiles. I have several computers and this is one of my required extensions :blue_heart: However, with sync enabled, my rules don't make sense between a 2160p and 1080p monitor.

**Comment by Enchufadoo** [comment]
@saildata can you explain what you mean by profiles? Or if there an extension that implements something similar so I can take a look.

**Comment by saildata** [comment]
> @saildata can you explain what you mean by profiles? Or if there an extension that implements something similar so I can take a look.

Yes. I use Firefox profiles / sync for different computers, so that my bookmarks, addons, etc are saved. I use the same Firefox profile on two different computers, one with a 2160p UHD monitor and the other with a 1080p monitor. The settings from fixedzoom are synced, and since I use a lot of the same sites, with cutom zoom levels, they are off when I change computers. 

On the [ArchWiki](https://wiki.archlinux.org/) I use 110% on UHD monitor and 100% on 1080p. I can't think of another extension that does this in particular, and especially not for zoom settings. In a perfect world, there would be a button on the advanced settings that allowed you to manage "profiles". For example, mine might be "UHD" and "HD", which would toggle between which set of rules I wanted to use at that time. Then when I switched computers, I would only have to make one change instead of manually zooming in/out a lot.

**Comment by OvidiuZeicu** [comment]
Is there a way of defining a zoom level to match screen pixel width? Not sure if that make sense, so I'll try to explain: I need a level of zoom that will match different screens resolutions so I can test website's layout. Similar with Dev Tools responsive mode, but inverted, for bigger resolutions, not smaller. Doing that in Dev Tools makes the responsive window bigger than the actual screen, and I have to keep scrolling around.
Thank you.

**Comment by Enchufadoo** [comment]
You can insert a script before the page loads, get the dpi like this site for example: http://dpi.lv/ and then make a zoom change.

It´s too slow for this extension, if the web api had a way of knowing this I would implement it but as for now it is not possible.

It´s a good idea for a new extension I guess.
